main/main.cpp

```diff
@@ -3660,6 +3671,15 @@ bool Main::iteration() {
 	process_max = MAX(process_ticks, process_max);
 	uint64_t frame_time = OS::get_singleton()->get_ticks_usec() - ticks;
 
+	if (print_fps_first_frame_ticks == 0) {
+		print_fps_first_frame_ticks = Time::get_singleton()->get_ticks_usec();
+	}
+	print_fps_last_frame_ticks = Time::get_singleton()->get_ticks_usec();
+
+	if (print_fps && Engine::get_singleton()->frames_drawn > PRINT_FPS_DISCARDED_FRAMES) {
+		print_fps_frametimes.push_back(frame_time);
+	}
```

Comment:
All this should be if `if (print_fps)` so that it doesn't impact iteration performance when not printing fps.